pyigm/clustering/xcorr_utils.py

```diff
@@ -352,23 +583,38 @@ def random_gal(galreal, Nrand, magbins, SPL):
 
     # TODO -- Use better masking than +/- 90 mag
     # Awful for loop, but it appears to run fast
+    minmag = np.min(magbins)
+    maxmag = np.max(magbins)
     for i in range(len(galreal)):
         if (galreal.MAG[i] > 90) or (galreal.MAG[i] < -90):  # no magnitude, use the whole distribution
             #vals = VALS['all']
             spl = SPL['all']
+        elif galreal.MAG[i] > maxmag:
+            spl = SPL['{}'.format(maxmag)]
+        elif galreal.MAG[i] < minmag:
+            spl = SPL['{}'.format(minmag)]
         else:
             # And this is ugly expensive
-            ind_mag = np.where(np.fabs(galreal.MAG[i] - magbins) == np.min(np.fabs(galreal.MAG[i] - magbins)))[0][0]
+            #ind_mag = np.where(np.fabs(galreal.MAG[i] - magbins) == np.min(np.fabs(galreal.MAG[i] - magbins)))[0][0]
+            try:
+                ind_mag = np.where((galreal.MAG[i] > magbins) & (galreal.MAG[i] < np.roll(magbins,-1)))[0][0]
+            except:
+                pdb.set_trace()
```

Comment:
please add a comment on when this could happen.

Reply:
added